NickvisionTubeConverter.Shared/Models/MediaUrlInfo.cs

```diff
@@ -107,40 +112,72 @@ private MediaUrlInfo(string url)
         });
     }
 
+    /// <summary>
+    /// Parses formats from media info
+    /// </summary>
+    private void ParseFormats(PyDict mediaInfo)
+    {
+        if(mediaInfo.HasKey("formats"))
+        {
+            foreach (var f in mediaInfo["formats"].As<PyList>())
+            {
+                var format = f.As<PyDict>();
+                if (format.HasKey("vbr"))
+                {
+                    var resolution = new VideoResolution(format["width"].As<int>(), format["height"].As<int>());
+                    if (!VideoResolutions.Contains(resolution))
+                    {
+                        VideoResolutions.Add(resolution);
+                    }
+                }
+            }
+            VideoResolutions.Sort((a, b) => b.CompareTo(a));
+        }
+    }
+
     /// <summary>
     /// Adds data retrieved from yt-dlp to MediaInfo
     /// </summary>
+    /// <param name="yt">The YouttubeDL object</param>
     /// <param name="mediaInfo">Python dictionary with media info</param>
     /// <param name="isPartOfPlaylist">Whether or not the media is part of a playlist</param>
-    private void ParseFromPyDict(PyDict mediaInfo, bool isPartOfPlaylist = false)
+    /// <param name="defaultUrl">A default URL to use if none available</param>
+    private void ParseFromPyDict(dynamic yt, PyDict mediaInfo, bool isPartOfPlaylist, string defaultUrl)
     {
         var title = mediaInfo.HasKey("title") ? (mediaInfo["title"].As<string?>() ?? "Media") : "Media";
         foreach (var c in Path.GetInvalidFileNameChars())
         {
             title = title.Replace(c, '_');
         }
-        foreach (var f in mediaInfo["formats"].As<PyList>())
+        if(mediaInfo.HasKey("formats"))
```

Comment:
This condition is repeated in `ParseFormats`

Reply:
Fixed in d0392cdcd02e3e94c213cdd417f23ba8f1b1f1a2